demisto_sdk/commands/common/hook_validations/graph_validator.py

```diff
@@ -368,3 +370,44 @@ def validate_unique_script_name(self):
                     is_valid = False
 
         return is_valid
+
+    @error_codes("GR108")
+    def validate_hidden_packs_do_not_have_mandatory_dependencies(self):
+        """
+        Validate that hidden pack(s) do not have dependant packs which the
+        hidden pack is a mandatory dependency for them.
+        """
+        is_valid = True
+
+        if dependant_packs := self.graph.find_mandatory_hidden_packs_dependencies(
+            pack_ids=self.pack_ids
+        ):
+            hidden_pack_id_to_dependant_pack_ids: dict = {
+                pack_id: set() for pack_id in self.pack_ids
+            }
+            for pack in dependant_packs:
+                for relationship in pack.depends_on:
+                    hidden_pack_id = relationship.content_item_to.object_id
+                    if hidden_pack_id in self.pack_ids:
```

Comment:
why do we need this condition?

Reply:
isn't there a possibility that the relationship wouldn't pick up something that is not a pack? if this is the case you are right, this condition is redundant